src/java/org/apache/cassandra/service/accord/AccordSyncPropagator.java

```diff
@@ -304,7 +344,22 @@ public boolean invokeOnFailure()
             {
                 return true;
             }
-        });
+        };
+        if (!failureDetector.isAlive(toEp))
+        {
+            // was the endpoint removed from membership?
+            ClusterMetadata metadata = ClusterMetadata.current();
+            if (Gossiper.instance.getEndpointStateForEndpoint(toEp) == null && !metadata.directory.allJoinedEndpoints().contains(toEp) && !metadata.fullCMSMembers().contains(toEp))
```

Comment:
was thinking it would be good to refactor as this was copy/pasted from `FailureDetector` but wasn't sure where to put it...